pkg/operator/ceph/object/shared_pools.go

```diff
@@ -0,0 +1,695 @@
+package object
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"reflect"
+	"sort"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	defaultPlacementName           = "default"
+	defaultPlacementCephConfigName = "default-placement"
+	defaultPlacementStorageClass   = "STANDARD"
+)
+
+func IsNeedToCreateObjectStorePools(sharedPools cephv1.ObjectSharedPoolsSpec) bool {
+	for _, pp := range sharedPools.PoolPlacements {
+		if pp.Name == defaultPlacementName {
+			// No need to create pools. External pools from default placement will be used
+			return false
+		}
+	}
+	if sharedPools.MetadataPoolName != "" && sharedPools.DataPoolName != "" {
+		// No need to create pools. Shared pools will be used
+		return false
+	}
+	return true
+}
+
+func validatePoolPlacements(placements []cephv1.PoolPlacementSpec) error {
+	names := make(map[string]struct{}, len(placements))
+	for _, p := range placements {
+		if _, ok := names[p.Name]; ok {
+			return fmt.Errorf("invalidObjStorePoolCofig: invalid placement %s: placement names must be unique", p.Name)
+		}
+		names[p.Name] = struct{}{}
+		if err := validatePoolPlacementStorageClasses(p.StorageClasses); err != nil {
+			return fmt.Errorf("invalidObjStorePoolCofig: invalid placement %s: %w", p.Name, err)
+		}
+	}
+	return nil
+}
+
+func validatePoolPlacementStorageClasses(scList []cephv1.PlacementStorageClassSpec) error {
+	names := make(map[string]struct{}, len(scList))
+	for _, sc := range scList {
+		if sc.Name == defaultPlacementStorageClass {
+			return fmt.Errorf("invalid placement StorageClass %s: %s name is reserved", sc.Name, defaultPlacementStorageClass)
+		}
+		if _, ok := names[sc.Name]; ok {
+			return fmt.Errorf("invalid placement StorageClass %s: name must be unique", sc.Name)
+		}
+		names[sc.Name] = struct{}{}
+	}
+	return nil
+}
+
+func sharedPoolsExist(objContext *Context, sharedPools cephv1.ObjectSharedPoolsSpec) error {
+	existingPools, err := cephclient.ListPoolSummaries(objContext.Context, objContext.clusterInfo)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list pools")
+	}
+	existing := make(map[string]struct{}, len(existingPools))
+	for _, pool := range existingPools {
+		existing[pool.Name] = struct{}{}
+	}
+	// ignore pools with empty name:
+	existing[""] = struct{}{}
```

Comment:
I'm somewhat confused about this. IIUC, this is helping to allow `sharedPools.MetadataPoolName`, `sharedPools.DataPoolName`, or `pp.DataNonECPoolName` to be empty below. Is that right? 

Could you update the comment to be a little more clear about this, and why?